File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

class Gigasecond {

    private static final Long GIGA_SECOND = 1_000_000_000L;
    private final LocalDateTime dateTime;

    Gigasecond(LocalDate moment) {
        this(moment.atStartOfDay());
    }

    Gigasecond(LocalDateTime moment) {
        dateTime = moment.plusSeconds(GIGA_SECOND);
    }

    LocalDateTime getDateTime() {
        return dateTime;
    }

}

File: scala/gigasecond/src/main/scala/Gigasecond.scala
import java.time.{LocalDate, LocalDateTime, LocalTime}

object Gigasecond {

  private val GigaSecond: Long = 1000000000L

  def add(startDate: LocalDate): LocalDateTime = LocalDateTime.of(startDate, LocalTime.MIDNIGHT).plusSeconds(GigaSecond)

  def add(startDateTime: LocalDateTime): LocalDateTime = startDateTime.plusSeconds(GigaSecond)
}
